### PR TITLE
readline: conan v2 support

### PR DESCRIPTION
**Pull request by WilliamBehrens**
Specify library name and version:  **readline/***

Work the readline recipe for better conan v2 support

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by WilliamBehrens** [comment]
I used `self.source_folder` as a replacement for `self._source_subfolder` and i'm unsure if that was the right choice. It seems like a much better alterative would be to change the `_patch_sources` method into actual patch files

File: recipes/readline/all/conanfile.py [hunk 1]
```diff
@@ -41,55 +39,48 @@ def requirements(self):
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
-        del self.settings.compiler.libcxx
```
Comment (SpaceIm) [inline recipes/readline/all/conanfile.py:44]: why is it removed?

File: recipes/readline/all/conanfile.py [hunk 2]
```diff
@@ -41,55 +39,48 @@ def requirements(self):
...
-
-        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-        if self.settings.os == "Macos" and self.settings.arch == "armv8":
-            # this should be really done automatically by the AutoToolsBuildEnvironment helper,
-            # but very unlikely old helper will be fixed any time soon
-            self._autotools.link_flags.extend(["-arch", "arm64"])
-        self._autotools.configure(args=configure_args, configure_dir=self._source_subfolder)
-        return self._autotools
+            tc.configure_args.extend(["--enable-static", "--disable-shared"])
+        if cross_building(self.settings):
```
Comment (SpaceIm) [inline recipes/readline/all/conanfile.py:63]:
```suggestion
        if cross_building(self):
```

File: recipes/readline/all/conanfile.py [hunk 3]
```diff
@@ -41,55 +39,48 @@ def requirements(self):
...
-            configure_args.append("bash_cv_wcwidth_broken=yes")
-
-        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-        if self.settings.os == "Macos" and self.settings.arch == "armv8":
-            # this should be really done automatically by the AutoToolsBuildEnvironment helper,
-            # but very unlikely old helper will be fixed any time soon
-            self._autotools.link_flags.extend(["-arch", "arm64"])
-        self._autotools.configure(args=configure_args, configure_dir=self._source_subfolder)
-        return self._autotools
+            tc.configure_args.extend(["--enable-static", "--disable-shared"])
```
Comment (SpaceIm) [inline recipes/readline/all/conanfile.py:62]: useless with AutotoolsToolchain, it's already injected under the hood.

File: recipes/readline/all/conanfile.py [hunk 4]
```diff
@@ -41,55 +39,48 @@ def requirements(self):
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
```
Comment (SpaceIm) [inline recipes/readline/all/conanfile.py:41]:
```suggestion
            self.options.rm_safe("fPIC")
```

File: recipes/readline/all/conanfile.py [hunk 5]
```diff
@@ -41,55 +39,48 @@ def requirements(self):
...
-        del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
```
Comment (SpaceIm) [inline recipes/readline/all/conanfile.py:42]:
```suggestion
        self.settings.rm_safe("compiler.cppstd")
```

File: recipes/readline/all/conanfile.py [hunk 6]
```diff
@@ -1,7 +1,11 @@
...
-from conans import ConanFile, AutoToolsBuildEnvironment, tools
-from conans.errors import ConanInvalidConfiguration
-
+from conan import ConanFile
+from conan.tools.files import get, replace_in_file, rmdir
+from conan.tools.build import cross_building
+from conan.tools.layout import basic_layout
+from conan.tools.microsoft import is_msvc
+from conan.tools.gnu import AutotoolsToolchain, Autotools
+from conan.errors import ConanInvalidConfiguration
```
Comment (SpaceIm) [inline recipes/readline/all/conanfile.py:8]:
```suggestion
from conan.errors import ConanInvalidConfiguration

required_conan_version = ">=1.53.0"

```

File: recipes/readline/all/conanfile.py [hunk 7]
```diff
@@ -1,7 +1,11 @@
 import os
-from conans import ConanFile, AutoToolsBuildEnvironment, tools
-from conans.errors import ConanInvalidConfiguration
-
+from conan import ConanFile
+from conan.tools.files import get, replace_in_file, rmdir
```
Comment (SpaceIm) [inline recipes/readline/all/conanfile.py:3]:
```suggestion
from conan.tools.apple import fix_apple_shared_install_name
from conan.tools.files import get, replace_in_file, rmdir
```

File: recipes/readline/all/conanfile.py [hunk 8]
```diff
@@ -41,55 +39,48 @@ def requirements(self):
...
-        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
-        autotools = self._configure_autotools()
+        self.copy(pattern="COPYING", dst="licenses", src=self.source_folder)
+        autotools = Autotools(self)
         autotools.install()
 
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-        tools.rmdir(os.path.join(self.package_folder, "share"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
```
Comment (SpaceIm) [inline recipes/readline/all/conanfile.py:83]:
```suggestion
        rmdir(self, os.path.join(self.package_folder, "share"))
        fix_apple_shared_install_name(self)
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]

File: recipes/readline/all/conanfile.py [hunk 9]
```diff
@@ -41,55 +39,48 @@ def requirements(self):
...
     def build(self):
         self._patch_sources()
-        autotools = self._configure_autotools()
+        autotools = Autotools(self)
         autotools.make()
 
     def package(self):
-        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
-        autotools = self._configure_autotools()
+        self.copy(pattern="COPYING", dst="licenses", src=self.source_folder)
```
Comment (SpaceIm) [inline recipes/readline/all/conanfile.py:78]:
```suggestion
        self.copy(pattern="COPYING", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
```

File: recipes/readline/all/conanfile.py (hunk 1 above)
Comment (WilliamBehrens) [reply recipes/readline/all/conanfile.py:44]: I'm currently having to debug this on a Windows machine and the default profile generated by conan 2 doesn't have that option and thus throws an error when exporting. Forgot that the profile generated on unix platforms is different

Comment (SpaceIm) [reply recipes/readline/all/conanfile.py:44]: just replace by `self.settings.rm_safe("compiler.libcxx")`

**Comment by WilliamBehrens** [comment]
> [CLA assistant check](https://cla-assistant.io/conan-io/conan-center-index?pullRequest=16433) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/conan-io/conan-center-index?pullRequest=16433) before we can accept your contribution.**1** out of **2** committers have signed the CLA.✅ WilliamBehrens❌ William Behrens
> 
> **William Behrens** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/conan-io/conan-center-index?pullRequest=16433) it.

How do I fix this? I realize my local .gitconfig is wrong

**Comment by WilliamBehrens** [comment]
Closing this as the commit history is horribly broken

**Pull request closed without merging**